cep-??.md

```diff
@@ -0,0 +1,246 @@
+
+`MatchSpec` queries can be also specified with keyword arguments between square brackets. Key-value pairs can be delimited by comma, space, or comma+space. Value can optionally be wrapped in single or double quotes, but must be wrapped if `value` contains a comma, space, or equal sign. Their values can be quoted with single or double quotes. The accepted keys are:
+
+- `channel` (`str`): Name or URL of a channel
+- `subdir` (`str`): Identifier of the subdir (either platform-specific or `noarch`)
+- `version` (`str`): A version specifier
+- `build` (`str`): Build string or a glob match.
+- `build_number` (`int`): Number of the build
+- `md5` (`str`): MD5 hash of the artifact
+- `sha256` (`str`): SHA256 hash of the artifact
+
+These are also accepted but have reduced utility. Their usage is discouraged:
+
+- `url`
+- `track_features`
+- `features`
+- `license`
+- `license_family`
+- `fn`
+
+When both positional and keyword arguments are used, the keyword arguments override the positional information.
+
+## Canonical representation
+
+Since there are several ways of specifying the same information, a canonical representation is welcome. The currently accepted notation is:
+
+```
+(channel(/subdir)::)name(version(build))[key1='value 1',key2=value2]
+```
+
+where `()` indicate optional fields. The rules for constructing a canonical string
+representation are:
+
+1. `name` (i.e. "package name") is required, but its value can be `*`. Its position is always
+   outside the key-value brackets.
+2. If `version` is an exact version, it goes outside the key-value brackets and is prepended
+   by `==`. If `version` is a "fuzzy" value (e.g. `1.11.*`), it goes outside the key-value
+   brackets with the `.*` left off and is prepended by `=`. Otherwise `version` is included
+   inside key-value brackets.
+3. If `version` is an exact version, and `build` is an exact value, `build` goes outside
+   key-value brackets prepended by a `=`. Otherwise, `build` goes inside key-value brackets.
+   `build_string` is an alias for `build`.
+4. The `namespace` position is being held for a future conda feature.
+5. If `channel` is included and is an exact value, a `::` separator is used between `channel`
+   and `name`. `channel` can either be a canonical channel name or a channel url. In the
+   canonical string representation, the canonical channel name will always be used.
+6. If `channel` is an exact value and `subdir` is an exact value, `subdir` is appended to
+   `channel` with a `/` separator. Otherwise, `subdir` is included in the key-value brackets.
```

Comment:
How does this related to the label channels? e.g. `pytorch/label/nightly::libfaiss`? 
With the seperator logic this will be assumed to be a subdir.

Reply:
The logic in conda is to take the last component and compare it against known subdirs. As a result, channels cannot be named like subdirs. e. g. I can't register a channel named `linux-64`.